aidatlu/main/tlu.py

```diff
@@ -405,41 +428,38 @@ def run(self) -> None:
         if self.zmq_address:
             self.setup_zmq()
 
-        t = threading.Thread(target=self.handle_status)
-        t.start()
-        while run_active:
-            try:
-                current_event = self.pull_fifo_event()
-                try:
-                    if save_data and np.size(current_event) > 1:
-                        self.data_table.append(current_event)
-                    if self.stop_condition == True:
-                        raise KeyboardInterrupt
-                except:
-                    if KeyboardInterrupt:
-                        run_active = False
-                        t.do_run = False
-                        self.stop_run()
-                    else:
-                        # If this happens: poss. Hitrate to high for FIFO and or Data handling.
-                        self.log.warning("Incomplete Event handling...")
-
-            except KeyboardInterrupt:
-                run_active = False
-                t.do_run = False
-                self.stop_run()
+    def run_loop(self) -> None:
+        """A single instance of the run loop. In a TLU run this function needs to be called repeatedly.
 
-        # Cleanup of FIFO
+        Raises:
+            KeyboardInterrupt: The run loop can be interrupted when raising a KeyboardInterrupt.
+        """
         try:
-            while np.size(current_event) > 1:
-                current_event = self.pull_fifo_event()
+            current_event = self.pull_fifo_event()
+            try:
+                if self.save_data and np.size(current_event) > 1:
+                    self.data_table.append(current_event)
+                if self.stop_condition == True:
+                    raise KeyboardInterrupt
+            except:
+                if KeyboardInterrupt:
+                    self.run_active = False
+                else:
+                    # If this happens: poss. Hitrate to high for FIFO and or Data handling.
+                    self.log.warning("Incomplete Event handling...")
+
         except KeyboardInterrupt:
-            self.log.warning("Interrupted FIFO cleanup")
+            self.run_active = False
+
+    def stop_run_configuration(self) -> None:
+        """Cleans remaining FIFO data and closes data files and zmq connections after a run."""
+        # Cleanup of FIFO
+        _ = self.pull_fifo_event()
```

Comment:
I dont think you need to assign the result to a private variable, just calling it suffices in python. Although maybe not in an interactive shell.